File: frontend/src/types/game.ts
export enum GameStatus {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  COMPLETED = 'COMPLETED',
  DROPPED = 'DROPPED',
}

export interface Game {
  id: number;
  title: string;
  platform: string;
  genre?: string;
  developer?: string;
  releaseYear?: number;
  status: GameStatus;
  rating?: number;
  notes?: string;
  coverUrl?: string;
  createdAt: string;
  updatedAt: string;
}

export interface GameCreateRequest {
  title: string;
  platform: string;
  genre?: string;
  developer?: string;
  releaseYear?: number;
  status: GameStatus;
  rating?: number;
  notes?: string;
  coverUrl?: string;
}

export interface GameUpdateRequest extends Partial<GameCreateRequest> {}

export interface StatsDTO {
  totalGames: number;
  countByStatus: Record<GameStatus, number>;
  averageRating: number;
}

File: backend/src/main/java/com/example/gamelibrary/model/Game.java
package com.example.gamelibrary.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.OffsetDateTime;

@Entity
@Table(name = "games")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Game {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, length = 80)
    private String platform;

    @Column(length = 80)
    private String genre;

    @Column(length = 120)
    private String developer;

    @Column
    private Integer releaseYear;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private GameStatus status = GameStatus.NOT_STARTED;

    @Column
    private Integer rating;

    @Column(columnDefinition = "TEXT")
    private String notes;

    @Column(length = 500)
    private String coverUrl;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private OffsetDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private OffsetDateTime updatedAt;
}
